src/compiler/path.ts

```diff
@@ -625,7 +625,28 @@ export function getNormalizedPathComponents(path: string, currentDirectory: stri
 
 /** @internal */
 export function getNormalizedAbsolutePath(fileName: string, currentDirectory: string | undefined): string {
-    return getPathFromPathComponents(getNormalizedPathComponents(fileName, currentDirectory));
+    fileName = normalizeSlashes(fileName);
+
+    if (isNotNormalizedOrAbsolute(fileName)) {
+        return getPathFromPathComponents(getNormalizedPathComponents(fileName, currentDirectory));

```

Comment:
If `fileName` is already absolute, we could just call `normalizePath` on it, which has already been optimized.

I also wonder about both `getNormalizedAbsolutePath` and `normalizePath`, whether it’s better to index through the string, building a new one if needed, rather than doing the two array allocations involved in `getPathFromPathComponents(reducePathComponents(getPathComponents(path)))`?

Reply:
I absolutely agree on that part, I think that iterating through each non-normalized component would be way better.

Reply:
Narrator: it was way better

<img width="958" alt="image" src="https://github.com/user-attachments/assets/a7c0804e-8c00-4b45-9c67-94c92e31d288" />